tests/modules/programs/yazi/keymap-expected.toml

```
@@ -1,27 +1,27 @@
 [[input.keymap]]
-exec = "close"
+run = "close"
 on = ["<C-q>"]
```

Will have to reorder the keys to account for alphabetic ordering. I.e., doing

```suggestion
on = ["<C-q>"]
run = "close"
```

for all examples.

---

You can use `nix build -L .#test-yazi-settings` to check the test.

---

Okay I've made sure it's alphabetical and that the test passes. Just out of curiosity, how does it know to generate in alphabetical order?

---

I think the reason is that in Nix, all attributes are evaluated to an alphabetical order, no matter in what order you define them. For example:

```
nix-repl> test = { b = "world"; a = "hello"; }

nix-repl> test
{
  a = "hello";
  b = "world";
}
```

As such, the toml file generated from the keymap attrset contains keys which are sorted in an alphabetical order, which I assume is because the toml generator takes the inputs from first to last. In the example above, it starts with a, then b.

However, the keys in the expected toml file in the tests are not sorted, so a conflict occurs and the test fails.

---

That makes sense, thanks! I'm used to hashset orders being non-deterministic, I like this nix solution.